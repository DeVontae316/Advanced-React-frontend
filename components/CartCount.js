import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import{TransitionGroup, CSSTransition} from 'react-transition-group';

const Dot = styled.div`
background:${props => props.theme.red};
border-radius:50%;
padding:0.5rem;
line-height:2rem;
min-width:3rem;
margin-left:1rem;
font-weight:100;
font-feature-setting:'tnum';
font-variant-numeric:tabular-numeric;
.count{
    color:white;
    text-align:center;
}
`
const AnimationStyles = styled.span`
 position:relative;
 .count{
     display:block;
     position:relative;
     transition:all 0.2s;
     backface-visibility:hidden;

 }
 .count-enter{
     transform: scale(1) rotateX(0.4turn);
 }
 .count-enter-active{
    transform: rotateX(0);
}
.count-exit{
 top:10;
 positon:relative;
 transform:rotateX(0);
}
.count-exit-active{
    transform: scale(1) rotateX(0.4turn);
}
`
const CartCount = ({count})=>{
    return(
    <AnimationStyles>
        <TransitionGroup>
            <CSSTransition 
            unmountOnExit
            className="count" 
            classNames="count" 
            key={count}
            timeout={{enter:200,exit:200}}
            >
            <Dot>
                <div className="count">
                    {count}
                </div>
            </Dot>
            </CSSTransition>
        </TransitionGroup>
    </AnimationStyles>

    
    )
}



export default CartCount;