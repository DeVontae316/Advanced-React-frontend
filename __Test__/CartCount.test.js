import CartCount from '../components/CartCount';
import {shallow,mount} from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('<CartCount/>',()=>{
    it('Cart Count',()=>{
      shallow(<CartCount count={10} />)
    });
    it('Matches snapshot',()=>{
        const wrapper = shallow(<CartCount count={10} />)
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
    it('Does it update',()=>{
        const wrapper = shallow(<CartCount count={11} />)
        
        expect(toJSON(wrapper)).toMatchSnapshot();
        
        wrapper.setProps({count:50});
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
    

})