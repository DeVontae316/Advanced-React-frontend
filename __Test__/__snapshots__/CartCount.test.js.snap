// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> Does it update 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="11"
      timeout={
        Object {
          "enter": 200,
          "exit": 200,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        <div
          className="count"
        >
          11
        </div>
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<CartCount/> Does it update 2`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="50"
      timeout={
        Object {
          "enter": 200,
          "exit": 200,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        <div
          className="count"
        >
          50
        </div>
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<CartCount/> Matches snapshot 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="10"
      timeout={
        Object {
          "enter": 200,
          "exit": 200,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        <div
          className="count"
        >
          10
        </div>
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;
